### DISCUSSION TITLE
Deploying on CloudFlare Pages or Vercel

### DISCUSSION THREAD

**Discussion by AlessioGr** [payloadcms/payload#528] [Q&A] [answered]
Hey,
is it possible to deploy Payload CMS to either CloudFlare Pages or Vercel? If yes, could someone give me a very brief step by step guide?

Thank you,
Alessio

**Comment by jmikrut** [comment, marked as answer]
Hey @AlessioGr — we've actually been working with another community member about deploying to Vercel in the following Discussion:

https://github.com/payloadcms/payload/discussions/519#discussioncomment-2546637

TL;DR: Vercel is a serverless platform and Payload runs on Express. You _**can**_ deploy Express apps to Vercel, but they don't recommend it. Check out that link for more info.

Regarding Cloudflare Pages, I believe that platform is meant to serve _static_ sites, and Payload comes with APIs. So I'm not sure that you'd be able to deploy Payload to Cloudflare Pages.

**Reply by AlessioGr** [reply]
I see, thank you a lot for the explanation! As for Cloudflare, you can probably use "Workers", but I guess that has the similar limitations as Vercel serverless functions

**Reply by balupton** [reply]
So Cloudflare Pages is just some tooling around Cloudflare Workers and KV to make edge static and dynamic sites possible, it is Cloudflare's competitor to Vercel. So Remix and Astro sites deployed to Cloudflare Pages support static assets, ssr, edge functions, and client components; plus the benefits of cloudflare; worlds largest cdn and ddos protection, kv, d1, r2, queue, etc. That said, the Cloudflare suggestion is best suited by https://github.com/payloadcms/payload/discussions/1525

**Reply by balupton** [reply]
For facilitation of the various serverless platforms, swapping out express for https://github.com/honojs/hono might be a good move

**Reply by GerardPolloRebozado** [reply]
Now that cloudflare has made a package to support nextjs SSR is it posible to host payload cms?
https://developers.cloudflare.com/pages/framework-guides/nextjs/ssr/get-started/#existing-apps

**Comment by dipanshuhappy** [comment]
This is not possible because cloudflare only support edge runtime , and payload cms has dependency of nodemailer which does not work on edge runtime

**Comment by pawel-ddh** [comment]
Hi, now CloudFlare already supports “runntine node.js”, https://opennext.js.org/cloudflare#cloudflare the question is if there is a way to run PayloadCMS on CloudFlare Workers ? 

When I try to open the admin I get an error:
```
 ⨯ Error: [unenv] process.report.getReport is not implemented yet!
 ```
 
 does anyone know any way?

**Reply by Las-re** [reply]
#6110 should be more useful for you

**Reply by dvogeldev** [reply]
And now they have a template https://blog.cloudflare.com/payload-cms-workers/

**Reply by supfilistine** [reply]
I am actively trying that template and on the face of it, it works, but with a specific payloadcms plugin, runs into a "code generation not allowed in this context" error that cant be reproduced using local next.

https://www.npmjs.com/package/@ai-stack/payloadcms

I asked in CF Dev discord but does anyone know why CF throws this error in workers?
